### PR TITLE
[WIP]docs: remove v3 to v4 migration docs

### PR DESCRIPTION
**Pull request by huangkairan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [x] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     docs: remove v3 to v4 migration docs      |
| 🇨🇳 Chinese |   移除 v3 至 v4 的迁移文档        |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

**Comment by huangkairan** [comment]
auto-complete done

**Comment by huangkairan** [comment]
breadcrumb done

File: components/config-provider/__tests__/__snapshots__/components.test.tsx.snap [hunk 1]
```diff
@@ -42168,7 +42168,7 @@ exports[`ConfigProvider components TreeSelect prefixCls 1`] = `
             <div
               aria-hidden="true"
               class="prefix-TreeSelect-tree-treenode"
-              style="position: absolute; pointer-events: none; visibility: hidden; height: 0px; overflow: hidden; border: 0px; padding: 0px;"
+              style="position: absolute; pointer-events: none; visibility: hidden; height: 0px; overflow: hidden;"
```
Comment (afc163) [inline components/config-provider/__tests__/__snapshots__/components.test.tsx.snap:42171]: 别改这个，rebase 一下 master

**Pull request closed without merging**